Compiler/src/inferenceresult.jl

```diff
@@ -183,7 +183,8 @@ function cache_lookup(𝕃::AbstractLattice, mi::MethodInstance, given_argtypes:
     method = mi.def::Method
     nargtypes = length(given_argtypes)
     for cached_result in cache
-        cached_result.linfo === mi || @goto next_cache
+        cached_result.tombstone && continue # ignore deleted entries (due to LimitedAccuracy)
```

Comment:
FWIW: this is the bugfix, the rest is various improvements of various sorts

n.b. the LimitedAccuracy object does not need to contain a Set anymore, since all of the nodes now have a `frameid::Int`, which is reliable and strongly ordered, so we could now build the lattice over the frameid value instead, which may be much simpler.